src/docs/regulatory_predicted_molecular_consequence.md

```diff
@@ -0,0 +1,44 @@
+# Regulatory Predicted Molecular Consequence
+
+| Field             | Type            | Description
+|-------------------|-----------------|---------------------
+|allele_name|string|Data-derived allele name
+|stable_id|string|Feature stable id
+|feature_type|OntologyTermMetadata|SO term for feature type (eg. transcript regulatory feature)
+|sub_feature_type|OntologyTermMetadata|SO term for feature type (eg. enhancer, CTCF)
+|consequences|array of OntologyTermMetadata|SO terms for predicted impact ( eg regulatory_region_variant)
+|regulatory_feature_relative_location|VariantRelativeLocation or null|Relative location on regulatory feature
+
+
+```json
+"name": "rs1761454157",
+"alleles":  [
+    {
+    "name": "5:173235220:G:A",
+    "predicted_molecular_consequences": {},
+    "regulatory_predicted_molecular_consequences": {
+        "allele_name": "A",
+        "stable_id": "ENSR5_94JQHZ",
+        "feature_type": {
+            "value": "Regulatory Feature"
```

Comment:
Our general approach with SO is to use the most specific relevant term, so the feature type would be promotor, enhancer etc.

Reply:
The client would need this, or similar, field to know what data (i.e. what set of fields) is contained in this structure, and what display logic to run for it.

Ideally, such a field would be a string rather than an object with a `value` field, and it would uniquely identify the shape of the data. For example, if, from the the perspective of the data served by hypsipyle, promoters are indistinguishable from enhancers etc., it would be great to have a single string that identifies them, rather than different strings that all refer to the same data structure.

Also, while modelling the predicted molecular consequence, we should probably also consider the shape of the VEP results payload.

Here is what a predicted molecular consequence currently looks like for the client [when it comes from hypsipyle](https://github.com/Ensembl/ensembl-client/blob/main/src/shared/types/variation-api/variantPredictedMolecularConsequence.ts#L20-L30) (no feature type was yet specified) and [when it comes from VEP](https://github.com/Ensembl/ensembl-client/blob/main/src/content/app/tools/vep/types/vepResultsResponse.ts#L57-L71) (we improvised a feature type, setting it to `null` for intergenic consequences).

Reply:
Hi @azangru, I created a [document](https://www.ebi.ac.uk/seqdb/confluence/display/EV/Revisiting+predicted_molecular_consequences+structure) to list out the different approaches for handling predicted_molecular_consequence in variant allele.